File: packages/core/scripts/decide.ts
#!/usr/bin/env bun
// Emit "<STATUS>|<authorized verbs>" for a K1 kernel vector, so CI can compare
// @arky/core (TS) kernel decisions against arky-core (Rust).
// Usage: bun run scripts/decide.ts <path-to-k1-vector.json> <repo-root>
import { readFileSync } from 'node:fs';
import { evaluateKernel } from '../src/index.ts';

const path = process.argv[2];
const root = process.argv[3] ?? `${import.meta.dir}/../../..`;
const v = JSON.parse(readFileSync(path, 'utf-8'));
const c = v.inputs?.commitment;
if (!c) {
  process.stdout.write('NONE');
} else {
  const tims = [];
  if (v.context?.fixtures?.tim)
    tims.push(JSON.parse(readFileSync(`${root}/vectors/${v.context.fixtures.tim}`, 'utf-8')).tim);
  // Inline evidence (K2 vectors embed their TIMs directly so they are
  // self-contained); each entry is a full TIM object.
  if (Array.isArray(v.context?.evidence)) tims.push(...v.context.evidence);
  const d = evaluateKernel(c, tims, { time: v.context?.time });
  process.stdout.write(
    `${d.status}|${d.authorized.map((x: { name: string }) => x.name).join(',')}`,
  );
}

File: packages/core-py/tools/decide.py
"""Emit "<STATUS>|<authorized verbs>" for a K1/K2 kernel vector, so CI can
compare Python kernel decisions against the other stacks.

Usage: python tools/decide.py <path-to-kernel-vector.json>
"""

from __future__ import annotations

import pathlib
import sys

import arky_core as arky


def read(path: pathlib.Path) -> object:
    return arky.parse(path.read_text(encoding="utf-8"))


def main() -> int:
    if len(sys.argv) < 2:
        print("usage: decide.py <vector.json>", file=sys.stderr)
        return 2
    vector = read(pathlib.Path(sys.argv[1]))
    commitment, found = arky.path(vector, "inputs", "commitment")
    if not found:
        sys.stdout.write("NONE")
        return 0

    vectors_dir = pathlib.Path(__file__).resolve().parents[3] / "vectors"
    tims: list[object] = []
    fixture_path = arky.get_str(vector, "context", "fixtures", "tim")
    if fixture_path:
        tim, ok = arky.path(read(vectors_dir / fixture_path), "tim")
        if ok:
            tims.append(tim)
    # K2 vectors embed their evidence inline so they are self-contained.
    evidence, ok = arky.path(vector, "context", "evidence")
    if ok and isinstance(evidence, list):
        tims.extend(evidence)

    eval_time = arky.get_str(vector, "context", "time") or "2025-10-15T12:00:00Z"
    decision = arky.evaluate_kernel(commitment, tims, eval_time)
    sys.stdout.write(f"{decision.status}|{','.join(decision.authorized)}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
